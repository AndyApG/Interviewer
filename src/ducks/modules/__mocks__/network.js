/* eslint-env jest */

import uuidv4 from '../../../utils/uuid';

const { actionTypes, entityPrimaryKeyProperty } = jest.requireActual('../network');

const reducer = jest.fn(() => ({
  nodes: [],
  edges: [],
  ego: {
    [entityPrimaryKeyProperty]: uuidv4(),
  },
}));

export default reducer;

export {
  actionTypes,
};
