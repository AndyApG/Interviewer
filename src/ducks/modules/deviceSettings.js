import deviceDescription from '../../utils/DeviceInfo';

const SET_DESCRIPTION = 'SETTINGS/SET_DESCRIPTION';
const SET_INTERFACE_SCALE = 'SETTINGS/SET_INTERFACE_SCALE';
const TOGGLE_SETTING = 'SETTINGS/TOGGLE_SETTING';

const initialState = {
  description: deviceDescription(),
  useFullScreenForms: window.matchMedia('screen and (min-device-aspect-ratio: 8/5), (min-device-height: 1800px)').matches,
  useDynamicScaling: true,
  interfaceScale: 100,
};

export default function reducer(state = initialState, action = {}) {
  switch (action.type) {
    case SET_DESCRIPTION:
      return {
        ...state,
        description: action.description,
      };
    case SET_INTERFACE_SCALE:
      return {
        ...state,
        interfaceScale: action.scale,
      };
    case TOGGLE_SETTING:
      return {
        ...state,
        [action.item]: !state[action.item],
      };
    default:
      return state;
  }
}

const setDescription = description => ({
  type: SET_DESCRIPTION,
  description,
});

const setInterfaceScale = scale => ({
  type: SET_INTERFACE_SCALE,
  scale,
});

const toggleSetting = item => ({
  type: TOGGLE_SETTING,
  item,
});

const actionCreators = {
  setDescription,
  setInterfaceScale,
  toggleSetting,
};

const actionTypes = {
  SET_DESCRIPTION,
  SET_INTERFACE_SCALE,
  TOGGLE_SETTING,
};

export {
  actionCreators,
  actionTypes,
};
