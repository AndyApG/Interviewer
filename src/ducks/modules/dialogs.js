const REGISTER_DIALOG = 'REGISTER_DIALOG';
const UNREGISTER_DIALOG = 'UNREGISTER_DIALOG';
const TOGGLE_DIALOG = 'TOGGLE_DIALOG';
const OPEN_DIALOG = 'OPEN_DIALOG';
const CLOSE_DIALOG = 'CLOSE_DIALOG';

const initialState = [];

export default function reducer(state = initialState, action = {}) {
  switch (action.type) {
    case REGISTER_DIALOG:
      return [
        ...state.filter(modal => modal.name !== action.name),
        { name: action.name, open: false },
      ];
    case UNREGISTER_DIALOG:
      return state.filter(modal => modal.name !== action.name);
    case TOGGLE_DIALOG:
      return state.map((modal) => {
        if (modal.name !== action.name) { return modal; }
        return {
          ...modal,
          open: !modal.open,
        };
      });
    case OPEN_DIALOG:
      return state.map((modal) => {
        if (modal.name !== action.name) { return modal; }
        return {
          ...modal,
          open: true,
        };
      });
    case CLOSE_DIALOG:
      return state.map((modal) => {
        if (modal.name !== action.name) { return modal; }
        return {
          ...modal,
          open: false,
        };
      });
    default:
      return state;
  }
}

function toggleDialog(name) {
  return {
    type: TOGGLE_DIALOG,
    name,
  };
}

function openDialog(name) {
  return {
    type: OPEN_DIALOG,
    name,
  };
}

function closeDialog(name) {
  return {
    type: CLOSE_DIALOG,
    name,
  };
}

function registerDialog(name) {
  return {
    type: REGISTER_DIALOG,
    name,
  };
}

function unregisterDialog(name) {
  return {
    type: UNREGISTER_DIALOG,
    name,
  };
}

const actionCreators = {
  toggleDialog,
  openDialog,
  closeDialog,
  registerDialog,
  unregisterDialog,
};

const actionTypes = {
  TOGGLE_DIALOG,
  OPEN_DIALOG,
  CLOSE_DIALOG,
  REGISTER_DIALOG,
  UNREGISTER_DIALOG,
};

const modalNames = {
  EDIT_NODE: 'EDIT_NODE',
  ADD_NODE: 'ADD_NODE',
};

export {
  actionCreators,
  actionTypes,
  modalNames,
};
