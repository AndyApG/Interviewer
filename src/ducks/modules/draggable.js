const DRAG_START = 'DRAG_START';
const DRAG_STOP = 'DRAG_STOP';

const initialState = {
  isDragging: false,
};

export default function reducer(state = initialState, action = {}) {
  switch (action.type) {
      case DRAG_START:
        return {
          ...state,
          isDragging: true,
        }
        case DRAG_STOP:
          return {
            ...state,
            isDragging: false,
          }
    default:
      return state;
  }
};

function dragStart() {
  return {
    type: DRAG_START,
  }
};

function dragStop() {
  return {
    type: DRAG_STOP,
  }
};

const actionCreators = {
  dragStart,
  dragStop,
};

const actionTypes = {
  DRAG_START,
  DRAG_STOP,
};

export {
  actionCreators,
  actionTypes,
};
