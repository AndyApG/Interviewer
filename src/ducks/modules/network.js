import { reject, findIndex, isMatch, omit } from 'lodash';

import uuidv4 from '../../utils/uuid';

// Property name for the primary key for nodes
export const NodePrimaryKeyProperty = '_uid';

// Property name for node "model" properties
export const NodeAttributesProperty = 'attributes';

export const ADD_NODES = 'ADD_NODES';
export const REMOVE_NODE = 'REMOVE_NODE';
export const UPDATE_NODE = 'UPDATE_NODE';
export const TOGGLE_NODE_ATTRIBUTES = 'TOGGLE_NODE_ATTRIBUTES';
export const ADD_EDGE = 'ADD_EDGE';
export const TOGGLE_EDGE = 'TOGGLE_EDGE';
export const REMOVE_EDGE = 'REMOVE_EDGE';
export const SET_EGO = 'SET_EGO';
export const UNSET_EGO = 'UNSET_EGO';


// Initial network model structure
const initialState = {
  ego: {},
  nodes: [],
  edges: [],
};

function flipEdge(edge) {
  return { from: edge.to, to: edge.from, type: edge.type };
}

function edgeExists(edges, edge) {
  return (
    findIndex(edges, edge) !== -1 ||
    findIndex(edges, flipEdge(edge)) !== -1
  );
}

/**
 * existingNodes - Existing network.nodes
 * netNodes - nodes to be added to the network
 * additionalAttributes - node model properties
*/
function getNodesWithBatchAdd(existingNodes, newNodes, nodeAttributeData) {
  // Create a function to create a UUID and merge node attributes
  const withModelandAttributeData = newNode => ({
    [NodePrimaryKeyProperty]: uuidv4(),
    ...newNode, // second to allow existing UUID to be overwritten
    [NodeAttributesProperty]: {
      ...newNode[NodeAttributesProperty],
      ...nodeAttributeData,
    },
  });

  return existingNodes.concat(newNodes.map(withModelandAttributeData));
}

/**
 * @param {array} nodes - an array of objects representing nodes
 * @param {object} updatedNode - object representing the node to be updated and its new properties
 * @param {boolean} full - if the entire node should be overwritten
 */
function getUpdatedNodes(nodes, updatedNode, nodeAttributeData) {
  // Iterate over the nodes list
  const updatedNodes = nodes.map((node) => {
    // Skip nodes where the primary key doesn't match
    if (node[NodePrimaryKeyProperty] !== updatedNode[NodePrimaryKeyProperty]) { return node; }
    console.log('getUpdatedNodes');
    // if we have an attributes payload, merge with any existing attributes
    if (nodeAttributeData) {
      console.log('nodeAttributedata found');
      console.log(updatedNode, nodeAttributeData);
      return {
        ...node,
        ...updatedNode,
        [NodeAttributesProperty]: {
          ...node[NodeAttributesProperty],
          ...nodeAttributeData,
        },
        [NodePrimaryKeyProperty]: node[NodePrimaryKeyProperty],
      };
    }

    // If no attribute payload, just merge the new node data with the existing
    return {
      ...node,
      ...updatedNode,
      [NodePrimaryKeyProperty]: node[NodePrimaryKeyProperty],
    };
  });

  // Return the modified array of nodes
  return updatedNodes;
}

export default function reducer(state = initialState, action = {}) {
  switch (action.type) {
    case ADD_NODES: {
      return {
        ...state,
        nodes: getNodesWithBatchAdd(state.nodes, action.nodes, action.additionalAttributes),
      };
    }
    /**
     * TOGGLE_NODE_ATTRIBUTES
     * Used to toggle the value of a boolean variable between states when the node is tapped.
     */
    case TOGGLE_NODE_ATTRIBUTES: {
      // attributes = object containing node properties, minus _uid
      const attributes = omit(action.attributes, [NodePrimaryKeyProperty]);

      // Map over the nodes
      const updatedNodes = state.nodes.map((node) => {
        // Skip nodes with different primary keys
        if (node[NodePrimaryKeyProperty] !== action[NodePrimaryKeyProperty]) { return node; }

        //
        if (isMatch(node, attributes)) {
          return omit(node, Object.getOwnPropertyNames(attributes));
        }

        return {
          ...node,
          ...action.attributes,
        };
      });

      return {
        ...state,
        nodes: updatedNodes,
      };
    }
    case UPDATE_NODE: {
      return {
        ...state,
        nodes: getUpdatedNodes(state.nodes, action.node, action.additionalAttributes),
      };
    }
    case REMOVE_NODE: {
      const removeNodePrimaryKeyProperty = action[NodePrimaryKeyProperty];
      return {
        ...state,
        nodes: reject(state.nodes, node =>
          node[NodePrimaryKeyProperty] === removeNodePrimaryKeyProperty),
        edges: reject(state.edges, edge =>
          edge.from === removeNodePrimaryKeyProperty || edge.to === removeNodePrimaryKeyProperty),
      };
    }
    case ADD_EDGE:
      if (edgeExists(state.edges, action.edge)) { return state; }
      return {
        ...state,
        edges: [...state.edges, action.edge],
      };
    case TOGGLE_EDGE:
      // remove edge if it exists, add it if it doesn't
      if (edgeExists(state.edges, action.edge)) {
        return {
          ...state,
          edges: reject(reject(state.edges, action.edge), flipEdge(action.edge)),
        };
      }
      return {
        ...state,
        edges: [...state.edges, action.edge],
      };
    case REMOVE_EDGE:
      if (edgeExists(state.edges, action.edge)) {
        return {
          ...state,
          edges: reject(reject(state.edges, action.edge), flipEdge(action.edge)),
        };
      }
      return state;
    default:
      return state;
  }
}

/**
 * @namespace NetworkActionCreators
 */
const actionCreators = {};

const actionTypes = {
  ADD_NODES,
  UPDATE_NODE,
  TOGGLE_NODE_ATTRIBUTES,
  REMOVE_NODE,
  ADD_EDGE,
  TOGGLE_EDGE,
  REMOVE_EDGE,
  SET_EGO,
  UNSET_EGO,
};

export {
  actionCreators,
  actionTypes,
};
