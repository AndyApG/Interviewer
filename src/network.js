let namesList = ['Joshua', 'Bernie', 'Michelle', 'Gregory', 'Patrick', 'Barney', 'Jonathon','Myles','Alethia','Tammera','Veola','Meredith','Renee','Grisel','Celestina','Fausto','Eliana','Raymundo','Lyle','Carry','Kittie','Melonie','Elke','Mattie','Kieth','Lourie','Marcie','Trinity','Librada','Lloyd','Pearlie','Velvet','Stephan','Hildegard','Winfred','Tempie','Maybelle','Melynda','Tiera','Lisbeth','Kiera','Gaye','Edra','Karissa','Manda','Ethelene','Michelle','Pamella','Jospeh','Tonette','Maren','Aundrea','Madelene','Epifania','Olive'];

module.exports = class Network {

  constructor() {
      this._edges = [];
      this._nodes = [];
      this.reserved_ids = [];
  }

  addNode(properties) {

    if (!this.egoExists()) {
      throw new Error('You must add an Ego before attempting to add other nodes.');
    }

    let newNodeID = 0;

    while (this.getNode(newNodeID) !== false || this.reserved_ids.indexOf(newNodeID) !== -1) {
      newNodeID++;
    }

    let nodeProperties = {
      id: newNodeID
    };

    Object.assign(nodeProperties, properties); // ES6 approach to extending objects.

    this.nodes.push(nodeProperties);
    this.reserved_ids.push(newNodeID);
  }

  egoExists() {
    return true;
  }

  resetNetwork() {
    this.nodes = [];
    this.edges = [];
  }

  getNode(id) {
    // Ensure that ID is always treated as int for === comparisons to work reliably.
    id = parseInt(id);

    if (id === undefined) { return false; }
    for (var i = 0;i<this._nodes.length; i++) {
      if (this._nodes[i].id === id) {return this._nodes[i]; }
    }
    return false;

  }

  get nodes() {
    return this._nodes;
  }

};
