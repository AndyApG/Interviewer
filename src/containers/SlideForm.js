import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import { Scroller } from '../components';
import { entityAttributesProperty, entityPrimaryKeyProperty } from '../ducks/modules/network';
import Node from './Node';
import { Form } from './';

class SlideForm extends PureComponent {
  render() {
    const {
      form,
      updateNode,
      node,
      index,
    } = this.props;

    return (
      <div className="swiper-slide">
        <div className="slide-content">
          <Node {...node} />
          <div className="alter-form__form-container">
            <Scroller>
              <Form
                {...form}
                className="alter-form__form"
                initialValues={node[entityAttributesProperty]}
                controls={[]}
                autoFocus={false}
                form={`NODE_FORM_${index + 1}`}
                onSubmit={formData => updateNode(node[entityPrimaryKeyProperty], {}, formData)}
              />
            </Scroller>
          </div>
        </div>
      </div>
    );
  }
}

SlideForm.propTypes = {
  form: PropTypes.object,
  updateNode: PropTypes.func,
  node: PropTypes.object,
  index: PropTypes.number,
};

SlideForm.defaultProps = {
  form: {},
  index: 0,
  node: {},
  updateNode: () => {},
};

export default SlideForm;
