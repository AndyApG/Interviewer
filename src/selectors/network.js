import { findKey } from 'lodash';
import { getActiveSession, getCaseId } from './session';
import { createDeepEqualSelector } from './utils';
import { getProtocolCodebook } from './protocol';
import { asExportableNetwork, asWorkerAgentNetwork } from '../utils/networkFormat';
import { getEntityAttributes } from '../ducks/modules/network';

export const getNetwork = createDeepEqualSelector(
  (state, props) => getActiveSession(state, props),
  session => (session && session.network) || { nodes: [], edges: [] },
);

export const getNetworkNodes = createDeepEqualSelector(
  getNetwork,
  network => network.nodes,
);

export const getNetworkEgo = createDeepEqualSelector(
  getNetwork,
  network => network.ego,
);

export const getNetworkEdges = createDeepEqualSelector(
  getNetwork,
  network => network.edges,
);

export const getExportableNetwork = createDeepEqualSelector(
  getNetwork,
  (state, props) => getProtocolCodebook(state, props),
  (state, props) => getCaseId(state, props),
  (network, registry, _caseID) => asExportableNetwork(network, registry, { _caseID }),
);

export const getWorkerNetwork = createDeepEqualSelector(
  getNetwork,
  (state, props) => getProtocolCodebook(state, props),
  (network, registry) => asWorkerAgentNetwork(network, registry),
);

// The user-defined name of a node type; e.g. `codebook.node[uuid].name == 'person'`
export const makeGetNodeTypeDefinition = () => createDeepEqualSelector(
  (state, props) => getProtocolCodebook(state, props),
  (state, props) =>
    (props && props.type) ||
    (props && props.stage && props.stage.subject && props.stage.subject.type) ||
    (state && state.type),
  (codebook, nodeType) => {
    const nodeInfo = codebook && codebook.node;
    return nodeInfo && nodeInfo[nodeType];
  },
);

// See: https://github.com/codaco/Network-Canvas/wiki/Node-Labeling
const labelLogic = (codebookForNodeType, nodeAttributes) => {
  // In the codebook for the stage's subject, look for the displayVariable property
  // and try to retrieve this value as a key in the node's attributes.
  const displayVariable = codebookForNodeType && codebookForNodeType.displayVariable;
  if (displayVariable && nodeAttributes[displayVariable]) {
    return nodeAttributes[displayVariable];
  }

  // In the codebook for the stage's subject, look for a variable with a name
  // property of "name", and try to retrieve this value by key in the node's
  // attributes
  const variableCalledName = codebookForNodeType && codebookForNodeType.variables && findKey(codebookForNodeType.variables, ['name', 'name']);
  if (variableCalledName && nodeAttributes[variableCalledName]) {
    return nodeAttributes[variableCalledName];
  }

  // look for a property on the node with a key of ‘displayVariable’, and try to
  // retrieve this value as a key in the node's attributes.
  const nodeDisplayVariable = nodeAttributes.displayVariable;
  if (nodeDisplayVariable && nodeAttributes[nodeDisplayVariable]) {
    return nodeAttributes[nodeDisplayVariable];
  }

  // look for a property on the node with a key of ‘name’, and try to retrieve this
  // value as a key in the node's attributes.
  const nodeVariableCalledName = findKey(nodeAttributes, ['name', 'name']);
  if (nodeVariableCalledName) {
    return nodeAttributes[nodeVariableCalledName];
  }

  return 'No label';
};

// Gets the node label variable and returns its value, or "No label".
// See: https://github.com/codaco/Network-Canvas/wiki/Node-Labeling
export const makeGetNodeLabel = () => createDeepEqualSelector(
  (state, props) => {
    const getNodeTypeDefinition = makeGetNodeTypeDefinition(state, props);
    return getNodeTypeDefinition(state, props);
  },
  nodeTypeDefinition => node => labelLogic(nodeTypeDefinition, getEntityAttributes(node)),
);

export const makeGetNodeColor = () => createDeepEqualSelector(
  getProtocolCodebook,
  (_, props) => props.type,
  (codebook, nodeType) => {
    const nodeInfo = codebook.node;
    return (nodeInfo && nodeInfo[nodeType] && nodeInfo[nodeType].color) || 'node-color-seq-1';
  },
);

export const makeGetEdgeLabel = () => createDeepEqualSelector(
  getProtocolCodebook,
  (_, props) => props.type,
  (codebook, edgeType) => {
    const edgeInfo = codebook.edge;
    return (edgeInfo && edgeInfo[edgeType] && edgeInfo[edgeType].label) || '';
  },
);

export const makeGetEdgeColor = () => createDeepEqualSelector(
  getProtocolCodebook,
  (_, props) => props.type,
  (codebook, edgeType) => {
    const edgeInfo = codebook.edge;
    return (edgeInfo && edgeInfo[edgeType] && edgeInfo[edgeType].color) || 'edge-color-seq-1';
  },
);

export const makeGetNodeAttributeLabel = () => createDeepEqualSelector(
  getProtocolCodebook,
  (_, props) => props.subject.type,
  (_, props) => props.variableId,
  (codebook, nodeType, variableId) => {
    const nodeInfo = codebook.node;
    const variables = (nodeInfo && nodeInfo[nodeType] && nodeInfo[nodeType].variables) || {};
    return (variables && variables[variableId] && variables[variableId].label) || [];
  },
);

export const makeGetCategoricalOptions = () => createDeepEqualSelector(
  (state, props) => getProtocolCodebook(state, props),
  (_, props) => props.subject.type,
  (_, props) => props.variableId,
  (codebook, nodeType, variableId) => {
    const nodeInfo = codebook.node;
    const variables = (nodeInfo && nodeInfo[nodeType] && nodeInfo[nodeType].variables) || {};
    return (variables && variables[variableId] && variables[variableId].options) || [];
  },
);
