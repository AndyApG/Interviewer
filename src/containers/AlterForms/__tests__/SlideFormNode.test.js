/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import SlideForm from '../SlideFormNode';
import { entityAttributesProperty } from '../../../ducks/modules/network';

const mockProps = {
  form: {
    title: 'alpha',
    entity: 'node',
    type: 'person',
  },
  nodeIndex: 1,
  stageIndex: 1,
};

describe('<SlideFormNode />', () => {
  it('should render', () => {
    const component = shallow(<SlideForm {...mockProps} node={{ name: 'Bob' }} />);
    expect(component).toMatchSnapshot();
  });

  it('should render with prepopulated fields if provided', () => {
    const withInitialValues = {
      [entityAttributesProperty]: {
        foo: 'bar',
      },
    };
    const subject = shallow(<SlideForm {...mockProps} node={withInitialValues} />);
    expect(subject.find('.alter-form__form')).toHaveLength(1);
    expect(
      (subject.find('.alter-form__form')).prop('initialValues'),
    ).toEqual({ foo: 'bar' });
  });
});
