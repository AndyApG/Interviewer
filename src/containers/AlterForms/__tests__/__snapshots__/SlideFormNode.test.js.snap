// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SlideFormNode /> should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SlideForm
    form={
      Object {
        "entity": "node",
        "title": "alpha",
        "type": "person",
      }
    }
    index={0}
    node={
      Object {
        "name": "Bob",
      }
    }
    onUpdate={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="slide-content"
      >
        <Connect(Node)
          name="Bob"
        />
        <div
          className="alter-form__form-container"
        >
          <Scroller
            className=""
            onScroll={[Function]}
          >
            <Connect(AutoInitialisedForm)
              autoFocus={false}
              className="alter-form__form"
              controls={Array []}
              entity="node"
              form="NODE_FORM_1"
              initialValues={undefined}
              onSubmit={[Function]}
              title="alpha"
              type="person"
            />
          </Scroller>
        </div>
      </div>,
      "className": "swiper-slide",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Node)
            name="Bob"
          />,
          <div
            className="alter-form__form-container"
          >
            <Scroller
              className=""
              onScroll={[Function]}
            >
              <Connect(AutoInitialisedForm)
                autoFocus={false}
                className="alter-form__form"
                controls={Array []}
                entity="node"
                form="NODE_FORM_1"
                initialValues={undefined}
                onSubmit={[Function]}
                title="alpha"
                type="person"
              />
            </Scroller>
          </div>,
        ],
        "className": "slide-content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "name": "Bob",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Scroller
              className=""
              onScroll={[Function]}
            >
              <Connect(AutoInitialisedForm)
                autoFocus={false}
                className="alter-form__form"
                controls={Array []}
                entity="node"
                form="NODE_FORM_1"
                initialValues={undefined}
                onSubmit={[Function]}
                title="alpha"
                type="person"
              />
            </Scroller>,
            "className": "alter-form__form-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Connect(AutoInitialisedForm)
                autoFocus={false}
                className="alter-form__form"
                controls={Array []}
                entity="node"
                form="NODE_FORM_1"
                initialValues={undefined}
                onSubmit={[Function]}
                title="alpha"
                type="person"
              />,
              "className": "",
              "onScroll": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "autoFocus": false,
                "className": "alter-form__form",
                "controls": Array [],
                "entity": "node",
                "form": "NODE_FORM_1",
                "initialValues": undefined,
                "onSubmit": [Function],
                "title": "alpha",
                "type": "person",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="slide-content"
        >
          <Connect(Node)
            name="Bob"
          />
          <div
            className="alter-form__form-container"
          >
            <Scroller
              className=""
              onScroll={[Function]}
            >
              <Connect(AutoInitialisedForm)
                autoFocus={false}
                className="alter-form__form"
                controls={Array []}
                entity="node"
                form="NODE_FORM_1"
                initialValues={undefined}
                onSubmit={[Function]}
                title="alpha"
                type="person"
              />
            </Scroller>
          </div>
        </div>,
        "className": "swiper-slide",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(Node)
              name="Bob"
            />,
            <div
              className="alter-form__form-container"
            >
              <Scroller
                className=""
                onScroll={[Function]}
              >
                <Connect(AutoInitialisedForm)
                  autoFocus={false}
                  className="alter-form__form"
                  controls={Array []}
                  entity="node"
                  form="NODE_FORM_1"
                  initialValues={undefined}
                  onSubmit={[Function]}
                  title="alpha"
                  type="person"
                />
              </Scroller>
            </div>,
          ],
          "className": "slide-content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "Bob",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Scroller
                className=""
                onScroll={[Function]}
              >
                <Connect(AutoInitialisedForm)
                  autoFocus={false}
                  className="alter-form__form"
                  controls={Array []}
                  entity="node"
                  form="NODE_FORM_1"
                  initialValues={undefined}
                  onSubmit={[Function]}
                  title="alpha"
                  type="person"
                />
              </Scroller>,
              "className": "alter-form__form-container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Connect(AutoInitialisedForm)
                  autoFocus={false}
                  className="alter-form__form"
                  controls={Array []}
                  entity="node"
                  form="NODE_FORM_1"
                  initialValues={undefined}
                  onSubmit={[Function]}
                  title="alpha"
                  type="person"
                />,
                "className": "",
                "onScroll": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "autoFocus": false,
                  "className": "alter-form__form",
                  "controls": Array [],
                  "entity": "node",
                  "form": "NODE_FORM_1",
                  "initialValues": undefined,
                  "onSubmit": [Function],
                  "title": "alpha",
                  "type": "person",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
